### PR TITLE
MNW-228 Protect account deletion

### PR DESCRIPTION
**Pull request by NotEternal**
## Issues

- https://mnw.atlassian.net/browse/MNW-228

## Changes description

- Split `profile/remove_account/` code and add a password confirmation modal

## Preview

form

### CODE REVIEW

File: packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import Modal from '../../../../common/modal/Modal';
+import PasswordInput from '../../../../common/PasswordInput';
+import { StyledButtons, StyledTitle } from './ui';
+
+type PasswordConfirmationModalActions = {
+    onClose: VoidFunction;
+    onConfirm: VoidFunction;
+};
+
+const PasswordConfirmationModal: FC<PasswordConfirmationModalActions> = ({
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:16]: Why created a new component instead use of EnterPasswordForm?

File: packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { FC, useEffect, useState, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { currentTargetValue, forkEvent } from '../../../../../shared/lib/forms/selectors';
+import { isPasswordValid } from '../../../../accounts/EnterPasswordForm/lib/validate';
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:5]: cross-components import :(

File: packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    useEffect(() => {
+        setIsConfirmDisabled(!inLength(password));
+    }, [password]);
+
+    const [isPasswordError, setIsPasswordError] = useState(false);
+
+    const disablePasswordError = () => setIsPasswordError(false);
+
+    const checkPassword = useCallback(async () => {
+        if (!(await isPasswordValid(password))) {
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:33]: Same logic as in EnterPasswordForm

File: packages/frontend/src/components/profile/remove_account/ui/RemoveAccountModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    return (
+        <Modal id="remove-account-modal" isOpen={isOpen} onClose={onClose} modalSize="sm">
+            <StyledRemoveAccountModal>
+                <StyledTitle>{t('removeAccount.title')}</StyledTitle>
+                <StyledDescription>{t('removeAccount.desc')}</StyledDescription>
+                <label>
+                    <Checkbox
+                        checked={removeAccountDisclaimerApproved}
+                        onChange={(e) =>
+                            setRemoveAccountDisclaimerApproved(e.target.checked)
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/ui/RemoveAccountModal/index.tsx:40]: You can use `currentTargetChecked`

```
...
onChange={currentTargetChecked(setRemoveAccountDisclaimerApproved))}
...
```

File: packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx (hunk 1 above)
Comment (loosefer) [reply packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:16]: image

Comment (NotEternal) [reply packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:16]: I thought aboout it, but wasn't sure about condition changes.

File: packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx (hunk 2 above)
Comment (NotEternal) [reply packages/frontend/src/components/profile/remove_account/ui/PasswordConfirmationModal/index.tsx:5]: But if we'll use the the whole form it won't solve import problems 
`../../../../accounts/EnterPasswordForm`

File: packages/frontend/src/components/profile/remove_account/RemoveAccount.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                <Modal
+                    id="password-confirmation-on-remove-account"
+                    onClose={closeDeletionConfirmation}
+                    style={modalStyles}
+                    isOpen={isModalOpen}
+                    closeButton
+                >
+                    {showPasswordConfirmation && (
+                        <EnterPasswordForm
+                            title={t('enterPassword.removeAccountTitle')}
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/RemoveAccount.tsx:92]: cross components import

File: packages/frontend/src/components/accounts/EnterPasswordForm/EnterPassword.tsx [hunk 6]
```diff
@@ -67,6 +76,15 @@ const EnterPasswordForm: FC<EnterPasswordFormProps> = ({
             </Password>
 
             <Submit>
+                {onCancel && (
+                    <FormButton
+                        /*@ts-ignore*/
+                        color='light-gray-blue'
+                        onClick={onCancel}
+                    >
+                        {t('restoreAccount.cancelCaption')}
```
Comment (loosefer) [inline packages/frontend/src/components/accounts/EnterPasswordForm/EnterPassword.tsx:85]: cross components import

File: packages/frontend/src/components/profile/remove_account/RemoveAccount.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        }
+    };
+
+    const onPasswordConfirmation = () => {
+        openDeletionConfirmation();
+        closePasswordConfirmation();
+    };
+
+    const processAccountDeletion = async () => {
+        const walletAccounts = await wallet.removeWalletAccount(accountId);
```
Comment (loosefer) [inline packages/frontend/src/components/profile/remove_account/RemoveAccount.tsx:64]: let's move logic to ./lib from view, [like here](https://github.com/mynearwallet/near-wallet/pull/79/files#diff-d87c52dede03cb01cac01a4d435f85dc0d290555bf6b12bb5a9970a33bd551f0R22)

**Review by loosefer — CHANGES_REQUESTED** [review]

**Review by loosefer — APPROVED** [review]

**Pull request merged**